server/database.js

```diff
@@ -184,10 +184,18 @@
 
     /**
      * @typedef {string|undefined} envString
-     * @param {{type: "sqlite"} | {type:envString, hostname:envString, port:envString, database:envString, username:envString, password:envString}} dbConfig the database configuration that should be written
+     * @param {{type: "sqlite"} | {type:envString, hostname:envString, port:envString, database:envString, username:envString, password:envString, caFilePath:envString}} dbConfig the database configuration that should be written
      * @returns {void}
      */
     static writeDBConfig(dbConfig) {
+        // Move CA file to the data directory
+        if (dbConfig.caFilePath) {
+            const dataCaFilePath = path.join(Database.dataDir, "mariadb-ca.pem");
+            fs.renameSync(dbConfig.caFilePath, dataCaFilePath);
```

Comment:
An alternate approach shall be , using a generic CA bundle : Eg: Mozilla. Then provide an env variable to override it, either as a path or the ca bundle content itself.

Reply:
My first attempt at implementing this was to manage this entirely through environment variables but then I read from the contibuting.md the following:

> I personally do not like something that requires a lot of configuration before you can finally start the app. The goal is to make the Uptime Kuma installation as easy as installing a mobile app.

And

> Settings should be configurable in the frontend. Environment variables are discouraged, unless it is related to startup such as DATA_DIR

I believe this is something that should be configured through environment variables but if we want to respect the whishes of the maintainer, the user upload has to be there with all the complications it implies.

Reply:
Again just opinion :) 

If we look at various feature requests for external Db, I believe it was not implemented early to keep the simplicity you quoted. A user base who may use a dedicated DB would be certainly having the knowledge and willingness to get it done, by reading docs and Configure As Code , even if there are efforts involved. 

In that context, if we add a UI element, theoretically it is cluttering AND/OR complicating the experience of the user base who needs just out of the box experience. 

On the other hand, if we do it via configuration those who need it can implement in a CAC fashion, hopefully it is efficient for those user base. 

In a nutshell , if we do it via config, we may able to stretch the functionality without affecting the core slogan of simplicity.

Reply:
@louislam Would it be possible to have your input on this as I do agree with @vsmanu proposal.

Changes are minimal and simpler with an environment variable only configuration but this would only be possible if `Quote: Settings should be configurable in the frontend.` is not an hard requirement.